File: app/src/main/java/han/ayeon/searchimgandvideo/view/SearchTabPagerAdapter.java
package han.ayeon.searchimgandvideo.view;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class SearchTabPagerAdapter extends FragmentStatePagerAdapter {

    private final List<Fragment> fragmentList = new ArrayList<>();
    private final List<String> pagerTitle = new ArrayList<>();

    SearchTabPagerAdapter(FragmentManager fm){
        super(fm);
    }

    void addFragment(Fragment fragment, String title){
        fragmentList.add(fragment);
        pagerTitle.add(title);
    }

    public String getPageTitle(int position) {
        return pagerTitle.get(position);
    }


    @Override
    public Fragment getItem(int position) { return fragmentList.get(position); }

    @Override
    public int getCount() {
        return fragmentList.size() ;
    }

}

File: app/src/main/java/han/ayeon/searchimgandvideo/presentation/view/SearchTabPagerAdapter.kt
package han.ayeon.searchimgandvideo.presentation.view

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

/**
 * Created by HanAYeon on 2019-07-18.
 */

class SearchTabPagerAdapter(fragmentManager: FragmentManager) : FragmentStatePagerAdapter(fragmentManager) {

    private var fragmentList = ArrayList<Fragment>()
    private var pagerTitle = ArrayList<String>()


    fun addFragment(fragment: Fragment, title : String) {
        fragmentList.add(fragment)
        pagerTitle.add(title)
    }

    override fun getPageTitle(postion : Int) : String{
        return pagerTitle.get(postion)
    }


    override fun getItem(position: Int): Fragment {
        return fragmentList.get(position)
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

}